py/server/deephaven/table.py

```diff
@@ -70,23 +70,14 @@ def __init__(self, j_table: jpy.JType):
         self._definition = self.j_table.getDefinition()
         self._schema = None
 
-    def __repr__(self):
-        default_repr = super().__repr__()
+    def __str__(self):
+        default_str = super().__str__()
         column_dict = {col.name: col.data_type for col in self.columns[:10]}
-        repr_str = (
-            f"{default_repr[:-2]}, num_rows = {self.size}, columns = {column_dict}"
+        str_str = (
+            f"{default_str[:-2]}, num_rows = {self.size}, columns = {column_dict}"
         )
-        repr_str = repr_str[:115] + "...}>" if len(repr_str) > 120 else repr_str
-        return repr_str
-
-    def __eq__(self, table: Table) -> bool:
-        try:
-            if _JTableTools.diff(self.j_table, table.j_table, 1):
-                return False
-            else:
-                return True
-        except Exception as e:
-            raise DHError(e, "table equality test failed.") from e
```

Comment:
1. `==` will call into the java jni for `obj1 == obj2 || obj1.equals(obj2)`. (You can see in jpy_jobj.c / JObj_Equals). 
2. I *think* we are using the default java reference equality for Table. For other wrapped objects, it may be different. If we *really* want JTableTools.diff == 0 to be the definition of "equals", we should also make that change in java.
3. We should add a helper around JTableTools.diff if we think it's useful for users IMO.

Reply:
I'm happy to add some unit tests, but unfortunately our setup isn't well adjusted to attaching test-only java sources (which I'd like to do for testing abstract functionality against something like JObjectWrapper for example). We do have some experience attaching java test sources against *jpy*, but not against py-server afaict.

Reply:
So, it *is* possible to add to the test classpath accessible to py-server by modifying Integrations/src/test/java. It's not the prettiest, but it's functional.

Reply:
I agree.  If anything, we can consider a helper function.  I'm ok closing this set of comments, unless @jmao-denver thinks we need the helper function now.